File: (C++)ProblemSolving/ProblemSolving/문자열 압축.cpp
#include <string>
#include <vector>
#include <iostream>
using namespace std;

int solution(string s) {
	int answer = s.size();
	for (int i = 1; i <= s.size() / 2; i++) {
		string res_str = "";
		string stand_str = s.substr(0, i);
		int cnt = 1;
		for (int j = i; ; j += i) {
			string cmp_str;
			if (j < s.size()) {
				cmp_str = s.substr(j, i);
				if (stand_str == cmp_str) {
					cnt++;
					continue;
				}
			}
			if (cnt > 1) {
				res_str += to_string(cnt);
				cnt = 1;
			}
			res_str += stand_str;
			stand_str = cmp_str;
			if (j >= s.size()) break;
		}

		int tmp = res_str.size();
		if (tmp < answer) answer = tmp;
	}
	return answer;
}

File: (Pyton)ProblemSolving/ProblemSovling/카카오/문자열 압축.js
function solution(s) {
  var answer = "";
  for (var i = 1; i <= s.length; i++) {
    var compress_str = "";
    var count = 1;
    var stand_str = s.substr(0, i);
    for (var j = i; j + i <= s.length; j += i) {
      var cmp_str = s.substr(j, i);
      console.log("[] ", i, stand_str, " [] ", cmp_str);
      if (cmp_str === stand_str) count++;
      else {
        if (count > 1) {
          compress_str += toString(count);
          count = 1;
        }
        compress_str += stand_str;
        stand_str = cmp_str;
      }
    }
    if (count > 1) {
      compress_str += toString(count);
    }
    compress_str += stand_str;
    console.log(compress_str);
    if (!answer || compress_str.length < answer.length) answer = compress_str;
  }
  console.log(answer);
  return answer;
}

solution("sss");
